src/app/fdctl/config.c

```diff
@@ -354,7 +358,11 @@ config_parse_array( config_t * config,
     char * end = token + strlen( token ) - 1;
     while( FD_UNLIKELY( *end == ' ' ) ) end--;
     *(end+1) = '\0';
-    if( FD_LIKELY( end > token ) ) parse_key_value( config, section, key, token );
+    if( FD_LIKELY( end > token ) ) {
+      if( FD_UNLIKELY( !parse_key_value( config, section, key, token ) ) ) {
+        FD_LOG_ERR(( "unrecognized key %s", key ));
```

Comment:
Another nit but since this is a user configuration file, it would be good to have a long descriptive error message for this failure case. Eg,

"Error while parsing user configuration TOML file at `%s`. The configuration had an unrecognized key `[%s.%s]`.", file_path, section, key

(I know we don't do this for some other errors currently)

Reply:
Updated and added some test output to the description